src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -789,4 +789,10 @@ object messages {
     val msg = "unreachable code"
     val explanation = ""
   }
+
+  case class IllegalStartOfSimplePattern()(implicit ctx: Context) extends Message(32) {
+    val kind = "Syntax"
+    val msg = "illegal start of simple pattern"
+    val explanation = ""
```

Comment:
This message doesn't really add anything to justify its existence currently. Some things to consider for the explanation:

* When does this occur?
* What is a simple expression?
* Is there perhaps something that could be added to the message in order to help tooling / the user?

Let's try to put some semantics into this :)